src/backer/cardaning.py

```diff
@@ -72,43 +70,60 @@ def updateTip(self, tipHeight):
-                # Build transaction
-                builder = pycardano.TransactionBuilder(self.context)
-                builder.add_input_address(self.spending_addr)
-                builder.add_output(pycardano.TransactionOutput(self.spending_addr, pycardano.Value.from_primitive([TRANSACTION_AMOUNT])))
-                kel_data = bytes(kel)
-                # Chunk size
-                # bytearrays is not accept
-                value = [kel_data[i:i + 64] for i in range(0, len(kel_data), 64)]
-
-                # Metadata. accept int key type
-                builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
-
-                signed_tx = builder.build_and_sign([self.payment_signing_key],
-                                                change_address=self.spending_addr,
-                                                merge_change=True)
-
-                # Submit transaction
-                self.context.submit_tx(signed_tx.to_cbor())
-                transId = str(signed_tx.transaction_body.inputs[0].transaction_id)
-                trans = {
-                    "id": transId,
-                    "kel": kel.decode('utf-8'),
-                    "tip": self.tipHeight
-                }
-                self.keldbConfirming.pin(keys=transId, val=json.dumps(trans).encode('utf-8'))
+            if self.pending_kel:
+                kel_data = bytearray()
+                submitting_kel = []
+                trans_size = 0
+                tx_cbor = None
+
+                for kel_item in self.pending_kel[:]:
+                    kel_size = len(kel_item)
+
+                    if trans_size + kel_size >= MAX_TRANSACTION_SIZE:
+                        break
+
+                    # Build transaction
+                    builder = pycardano.TransactionBuilder(self.context)
+                    builder.add_input_address(self.spending_addr)
+                    builder.add_output(pycardano.TransactionOutput(self.spending_addr, pycardano.Value.from_primitive([TRANSACTION_AMOUNT])))
+
+                    submitting_kel.append(kel_item.decode('utf-8'))
+                    kel_data = kel_data + kel_item
+                    kel_data_bytes = bytes(kel_data)
+
+                    # Chunk size
+                    # bytearrays is not accept
+                    value = [kel_data_bytes[i:i + 64] for i in range(0, len(kel_data_bytes), 64)]
+
+                    # Metadata. accept int key type
+                    builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
+                    signed_tx = builder.build_and_sign([self.payment_signing_key],
+                                                    change_address=self.spending_addr,
+                                                    merge_change=True)
+                    tx_cbor = signed_tx.to_cbor()
+                    trans_size = len(tx_cbor)
+                    self.pending_kel.remove(kel_item)
+
+                if tx_cbor:
```

Comment:
When is `tx_cbor` None?

Reply:
It's never None in our case so I remove this condition :)